RFCs/0009-Fn::For.md

```diff
@@ -0,0 +1,903 @@
+        - Ipv6Address: "ipv6-2"
+        - Ipv6Address: "ipv6-3"
+```
+
+* Another example of using an outer loop’s variable inside inner loop
+```yaml
+Parameters:
+  Subnets:
+      Type: CommaDelimitedList
+      Default: "subnet1,subnet2"
+  TagValues:
+      Type: CommaDelimitedList
+      Default: "tag1,tag2"
+Resources:
+  Fn::Merge:
+    - Fn::Map:
+        index: i1
+        value: x1
+        collection: !Ref Subnets
+        key: !Sub "Instance${i1}"
+        fragment: 
+          InstanceType: "m1.small"
+          SubnetId: !Ref x1
+          Tags:
+            - Fn::Map:
+                index: i2
+                value: x2
+                collection: !Ref TagValues
+                fragment:
+                  Key: !Sub "${x1}"
+                  Value: !Sub "${x2}"
+```
+
+# Potential Follow-up Features
+
+The following features are out of scope of this RFC. Separate RFC could be created in the future if there is customer demand.
+
+### Indexing
+* You may want to reference a specific object among iteratively created objects in any place within template.
+```yaml
+Parameters:
+   Endpoints:
+     Type: CommaDelimitedList
+     Default: "endpoint1,endpoint2,endpoint3"
+Resources:
+  Topic:
+    Fn::Merge:
+      - Fn::Map:
+          collection: Endpoints
+          fragment:
+            Type: AWS::SNS::Topic
+            Properties:
+              Subscription:
+                - Endpoint: !Ref value
+Outputs:
+  TopicOutput:
+    Value: !Select [1, Topic]
+```
+
+### Modules
+* `Fn::Map` should support replicating Modules
+
+### Replicate a fragment for certain number of times
+* For a number parameter x, replicate a given template block x number of times.
+
+### Zipping
+* `Fn::Map` could take in multiple lists and generate an aggregated list of Resources or Outputs.
+
+# Limitations
+#### Replicate multiple Resource objects in a single Fn::Map usage
+* To replicate a Resource/Output object, a custom key interpolated with an index has to be provided under the `key` parameter. This means an `Fn::Map` loop can be used to replicate only one Resource at a time. To replicate multiple Resources, `Fn::Map` has to be defined for each Resource.
+
+#### A Collection to Iterate on has to have known values
+* All the values in a list have to be *known* before resource provisioning, otherwise it will throw an error message that Fn::Map received a value which cannot be resolved. You can’t, for example, refer to an attribute of a Resource. Values must be known before CFN performs any remote resource interactions.
+* For  example, the below would not work
+```yaml
+Resources:
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: '22'
+          ToPort: '22'
+          CidrIp: 0.0.0.0/0    
+  Instance:
+     Type: AWS::EC2::Instance
+     Properties:
+        SecurityGroups:
+          - Fn::Map:
+              collection: [!Ref SecurityGroup]
```

Comment:
`!GetAtt x.DnsEntries` of a  `AWS::EC2::VPCEndpoint` is also an interesting example